Homework3/task_2.py

```diff
@@ -0,0 +1,225 @@
+    return input_data
+
+
+def change_name_content(input_data):
+    my_logger.info('The content was changed in the field "name.title"')
+    return {'name.title': RULE_LIST.get(input_data['name.title'], input_data['name.title']), **input_data}
+
+
+def convert_to_format(input_data, field='dob.date', format_string='%Y-%m-%d'):
+    dob = datetime.strptime(input_data[field], '%Y-%m-%dT%H:%M:%S.%fZ')
+    input_data[field] = dob.strftime(format_string)
+    my_logger.info(f'Format of field {field} was changed to {format_string}')
+    return input_data
+
+
+def filter_data(input_data, gender=None, num_rows=None):
+    if gender:
+        input_data = [user for user in input_data if user['gender'] == gender]
+        # input_data = list(filter(lambda x: x['gender'] == gender, input_data))
+        my_logger.info('The data was filtered by gender')
+    elif num_rows and num_rows < len(input_data):
+        input_data = input_data[:num_rows]
+        my_logger.info('The data was filtered by num_rows')
+    else:
+        my_logger.warning('The data was not filtered by any of the parameters!!!')
+    return input_data
+
+
+def check_folder_existence(path):
+    os.makedirs(path, exist_ok=True)
+    my_logger.info(f'The folder was created along the path {path} ')
+    os.chdir(path)
+
+
+def move_file_to_destination(initial_file_path, destination_folder):
+    shutil.move(initial_file_path, destination_folder)
+    my_logger.info(f'The {initial_file_path} was moved to {destination_folder} ')
+
+
+def rearrange_data(input_data):
+    new_structure = {}
+    for data in input_data:
+        decade = f"{data['dob.date'][-2]}0-th"
+        country = data.pop('location.country')
+
+        new_structure.setdefault(decade, {}).setdefault(country, []).append(data)
+
+    my_logger.info('The data was rearranged')
+    return new_structure
+
+
+def create_filename(country_data, country_folder):
+    max_age = max([data['dob.age'] for data in country_data])
+    avg_registered = round(mean([int(data['registered.age']) for data in country_data]), 1)
+    most_common_id = Counter(([data['id.name'] for data in country_data])).most_common(1)[0][0]
+    filename = os.path.join(country_folder, f'{max_age}_{avg_registered}_{most_common_id}.csv')
+    return filename
+
+
+def create_subfolders(destination_folder, data):
+    for decade, countries in data.items():
+        for country, country_data in countries.items():
+            country_folder = os.path.join(destination_folder, decade, country)
+            os.makedirs(country_folder, exist_ok=True)
+
+            filename = create_filename(country_data, country_folder)
+            my_logger.info(f'Log paths {filename} was created.')
+
+            write_data_in_csv(filename, country_data)
+
+
+def remove_data_before_certain_decade(folder_path, decade):
+    contents = os.listdir(folder_path)
+    for f in contents:
+        if f[:2].isdigit() and int(f[:2]) < decade and int(f[:2]) != 0:
```

Comment:
could be refactored

Reply:
in python there is such type of conditions **< x <**
ex. 1 < x < 6. x should be less than 6 and greater than 1.

Reply:
f[:2] I believe it could be a variable